gateway-service/src/test/java/org/zowe/apiml/acceptance/ZosmfSchemeTest.java

```diff
@@ -9,62 +9,139 @@
 import org.zowe.apiml.acceptance.common.AcceptanceTestWithTwoServices;
 import org.zowe.apiml.acceptance.netflix.MetadataBuilder;
+import org.zowe.apiml.gateway.security.service.zosmf.ZosmfService;
+import org.zowe.apiml.gateway.utils.JWTUtils;
+import org.zowe.apiml.security.HttpsConfig;
 import org.zowe.apiml.util.config.SslContext;
 import org.zowe.apiml.util.config.SslContextConfigurer;
 
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
 /**
  * This test verifies that the token or client certificate was exchanged. The input is a valid apimlJwtToken/client certificate.
  * The output to be tested is the Zosmf token.
  */
 @AcceptanceTest
+@TestPropertySource(properties = {"apiml.security.auth.provider=zosmf", "spring.profiles.active=debug", "apiml.security.x509.externalMapperUrl="})
 class ZosmfSchemeTest extends AcceptanceTestWithTwoServices {
     @Value("${server.ssl.keyStorePassword:password}")
     private char[] keystorePassword;
     @Value("${server.ssl.keyStore}")
     private String keystore;
     private final String clientKeystore = "../keystore/client_cert/client-certs.p12";
+    @Value("${server.ssl.keyAlias:#{null}}")
+    private String keyAlias;
+
+    @Autowired
+    public ZosmfService zosmfService;
 
     @Nested
     class GivenClientCertificate {
         @BeforeEach
         void setUp() throws Exception {
             SslContextConfigurer configurer = new SslContextConfigurer(keystorePassword, clientKeystore, keystore);
             SslContext.prepareSslAuthentication(configurer);
-
             applicationRegistry.clearApplications();
+
             MetadataBuilder defaultBuilder = MetadataBuilder.defaultInstance();
             defaultBuilder.withZosmf();
             applicationRegistry.addApplication(serviceWithDefaultConfiguration, defaultBuilder, false);
             applicationRegistry.setCurrentApplication(serviceWithDefaultConfiguration.getId());
-
             reset(mockClient);
         }
 
         @Nested
         class WhenClientAuthInExtendedKeyUsage {
-            // TODO: add checks for transformation once X509 -> zosmf is implemented
-            @Test
-            void thenOk() throws IOException {
-                mockValid200HttpResponse();
-                given()
-                    .config(SslContext.clientCertUser)
-                    .when()
-                    .get(basePath + serviceWithDefaultConfiguration.getPath())
-                    .then()
-                    .statusCode(is(HttpStatus.SC_OK));
+
+            @Nested
+            class WhenZosmfAuthenticateResponseLTPA {
+
+                @BeforeEach
+                void setZosmfResponse() {
+                    Map<ZosmfService.TokenType, String> tokens = new HashMap<>();
```

Comment:
Can this preparation go to the WhenClientAuthInExtendedKeyUsage? 
As its used here and twenty lines later again.

Reply:
the issue here is that both are using different token - zosmf and zowe JWT. As JWT string is immutable, the tests will fail when the rest of the code is moved to a single method. The only part that can be moved is HttpConfig

Reply:
ok, not worth it then.